src/test/kotlin/edu/carroll/cs389/service/tracker/trackservice/TrackServiceGeneralPersistenceTest.kt

```diff
@@ -0,0 +1,154 @@
+
+    @Test
+    fun datastoreIsEmptyWithoutInsertTest() {
+        assertTrue(
+            "datastoreIsEmptyWithoutInsertTest(): test datastore (newly instantiated and cleaned after every test" +
+                    " should be empty without inserting data",
+            trackService.query().isEmpty()
+        )
+    }
+
+    @Test
+    fun singleInsertTest() {
+        val ipv4: String = "192.168.1.1"
+        val browser: String = "Firefox"
+        val browserMajorVersion: String = "110"
+        val platform: String = "Ubuntu"
+        val platformVersion: String = "22.04 LTS"
+        val requestURI: String = "/"
+
+        assertTrue(
+            "persistenceOfTwoInsertsTest(): should succeed with valid user agent",
+            trackService.trackClient(ipv4, browser, browserMajorVersion, platform, platformVersion, requestURI)
+        )
+
+        assertTrue(
+            "persistenceOfTwoInsertsTest(): inserted two records, should be of size 2",
+            trackService.query().size == 1
+        )
+
+        val user = trackService.query()[0]
+
+        assertTrue(
+            "persistenceOfTwoInsertsTest(): should query information from second insert first due to timestamp",
+            ipv4 == user.clientIpv4Address() &&
+                    "$browser $browserMajorVersion" == user.clientBrowserInfo() &&
+                    "$platform $platformVersion" == user.clientOperatingSystem() &&
+                    requestURI == user.clientConnectionRequestedPage()
+        )
+    }
+
+
+    @Test
+    fun persistenceOfTwoInsertsTest() {
+        val ipv41: String = "192.168.1.1"
+        val browser1: String = "Firefox"
+        val browserMajorVersion1: String = "110"
+        val platform1: String = "Ubuntu"
+        val platformVersion1: String = "22.04 LTS"
+        val requestURI1: String = "/"
+
+        val ipv42: String = "172.158.27.139"
+        val browser2: String = "Chrome"
+        val browserMajorVersion2: String = "37"
+        val platform2: String = "Windows"
+        val platformVersion2: String = "10 NT"
+        val requestURI2: String = "/ip_info"
+
+        assertTrue(
+            "persistenceOfTwoInsertsTest(): should succeed with valid user agent",
+            trackService.trackClient(ipv41, browser1, browserMajorVersion1, platform1, platformVersion1, requestURI1)
+        )
+
+        assertTrue(
+            "persistenceOfTwoInsertsTest(): should succeed with valid user agent",
+            trackService.trackClient(ipv42, browser2, browserMajorVersion2, platform2, platformVersion2, requestURI2)
+        )
+
+        assertTrue(
+            "persistenceOfTwoInsertsTest(): inserted two records, should be of size 2",
+            trackService.query().size == 2
+        )
+
+
+        // assuming that queryDatabase works correctly (must assume something works)
+        // we know the list is ordered by descending timestamp
+        val results = trackService.query()
+        val user1 = results[1]
```

Comment:
One database query, then using the results after.  Good.  Never-mind, you checked the size above.  The assignment should be made before the first assert.

Not verifying that there are two results before grabbing the two results, not so much.

     assertEquals(results.size(), 2)

(Done above, ...)